src/graph_generator.cc

```diff
@@ -141,7 +151,8 @@ namespace detail {
 			reduce_execution_front_to(epoch);
 		}
 
-		min_epoch_for_new_commands = min_new_epoch;
+		min_epoch_for_new_commands = epochs[0]->get_cid();
```

Comment:
Is this intentionally `epochs[0]` for all nodes? If so, why?

Reply:
This is the global minimum epoch command id (as opposed to the per-node epoch, which is set in `set_epoch_for_new_commands`). The vector is ordered by command id, as is verified by the assertion on the following line.